scripts/imgtool/image.py

```diff
@@ -69,8 +74,18 @@ class Image():
     @classmethod
     def load(cls, path, included_header=False, **kwargs):
         """Load an image from a given file"""
-        with open(path, 'rb') as f:
-            payload = f.read()
+        ext = os.path.splitext(path)[1][1:].lower()
+        if ext == INTEL_HEX_EXT:
+            ih = IntelHex(path)
+            payload = ih.tobinarray()
+            segments = ih.segments()
```

Comment:
Any chance that the linker might actually generate multiple segments? What about code and initialized data?

Reply:
Is it hard to support multiple segments? I don't know about any case which will require multiple segments - but i can imagine that a zephyr ARMv8 future target might  generates separate segments: 1st for secure and 2nd for non-secure part

Reply:
I have never seen multiple segments before in .hex files. Signing them separately would require deciding first on a format for multiple images and their signatures, which is outside the scope of this PR. What I can do is fail instead of warn because multiple segments at this point are unhandled.

Reply:
understand this - I seen multiple segments many times - but it was for complex systems with extensive usage of MPU. 
I agree - this is out of scope.